services/types.go

```diff
@@ -326,3 +330,22 @@ type HeartbeatService interface {
 	// Watch watches the heartbeats for a service
 	Watch() (xwatch.Watch, error)
 }
+
+// LeaderService provides access to etcd-backed leader elections.
+type LeaderService interface {
+	// Close closes the election service client entirely. No more campaigns can be
+	// started and any outstanding campaigns are closed.
+	Close() error
+
+	// Campaign proposes that the caller become the leader, blocking until it is
+	// elected.
+	Campaign() error
```

Comment:
We need to support campaigning for multiple locks, for example: if we want to support master/slave mode per shard, then a box might be owning 20 shards, and became the leader for 15 shards and slave for 5 shards. We should allow services to content for any keys under the service prefix, e.g: there could be _ld/env/svc/1, _ld/env/svc/2 (for shards), _ld/env/svc/bucket1 (bucket1 here means a subset of shards, as long as the service understands what it means), a service can create any lock under the service prefix.

Also I wonder if we should name the functions as lock/unlock to fit better into that semantic.